File: arsh_goyal_sde_sheet/arrays/283. Move Zeroes/rolling_snow_ball.py
# inspired by post https://leetcode.com/problems/move-zeroes/discuss/172432/THE-EASIEST-but-UNUSUAL-snowball-JAVA-solution-BEATS-100-(O(n))-+-clear-explanation

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        ballSize = 0
        N = len(nums)
        
        for i in range(N):
            if nums[i] == 0:
                ballSize += 1
            else:
                nums[i], nums[i-ballSize] = 0, nums[i]

File: arsh_goyal_sde_sheet/arrays/283. Move Zeroes/rolling_snow_ball.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int ballSize = 0, N = nums.size();
        for(int i = 0; i < N; i++) {
            if(nums[i] == 0) ballSize++;
            else {
                int temp = nums[i];
                nums[i] = 0;
                nums[i-ballSize] = temp;
            }
        }
    }
};